### PR TITLE
separate installer build and publish to separate jobs

### PR DESCRIPTION
**Pull request by altendky**
### Draft For:
- [x] review timeouts on publish jobs
- [x] ~review `setjobenv` usage for what is needed in which jobs~
- [x] ~can we use a more, or entirely, common image for the publish portion?~
  - great idea!  for later...
- [x] can the test jobs depend directly on the build job instead of publish?

### CODE REVIEW

File: .github/workflows/build-linux-installer-deb.yml [hunk 1]
```diff
@@ -212,6 +197,80 @@ jobs:
...
+  publish:
+    name: Publish ${{ matrix.os.arch }}
+    runs-on: ${{ matrix.os.runs-on }}
+    needs:
+    - build
+    container: chianetwork/ubuntu-20.04-builder:latest
+    timeout-minutes: 5
+    strategy:
+      fail-fast: false
+      max-parallel: 4
```
Comment (cmmarslender) [inline .github/workflows/build-linux-installer-deb.yml:213]:
```suggestion
```
Assume this was a copy/paste, but generally we dont need to limit matrices in cases like this (Looking earlier in the file, I now see this was already there for the other matrix.. might make sense to just get rid of the max-parallel for all the installer workflows)

File: .github/workflows/build-linux-installer-rpm.yml [hunk 2]
```diff
@@ -203,6 +190,74 @@ jobs:
...
+    name: Publish amd64 RPM
+    runs-on: ubuntu-latest
+    needs:
+    - build
+    container:
+      image: chianetwork/rocky8-builder:latest
+    timeout-minutes: 5
+    strategy:
+      fail-fast: false
+      max-parallel: 4
```
Comment (cmmarslender) [inline .github/workflows/build-linux-installer-rpm.yml:207]:
```suggestion
```
Same comment as before.. We should probably remove the other max-parallel in this file as well.

File: .github/workflows/build-macos-installers.yml [hunk 3]
```diff
@@ -250,6 +237,88 @@ jobs:
...
+
+  publish:
+    name: Publish ${{ matrix.os.name }} DMG
+    runs-on: ${{ matrix.os.runs-on }}
+    needs:
+    - build
+    timeout-minutes: 5
+    strategy:
+      fail-fast: false
+      max-parallel: 4
```
Comment (cmmarslender) [inline .github/workflows/build-macos-installers.yml:252]:
```suggestion
```
Same comment as before.. We should probably remove the other max-parallel in this file as well.

File: .github/workflows/build-macos-installers.yml [hunk 4]
```diff
@@ -250,6 +237,88 @@ jobs:
...
+        uses: actions/download-artifact@v3
+        with:
+          name: chia-installers-macos-dmg-${{ matrix.os.name }}
+          path: build_scripts/final_installer/
+
+      - name: Set Env
+        uses: Chia-Network/actions/setjobenv@main
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          MACOSX_DEPLOYMENT_TARGET: 11
```
Comment (cmmarslender) [inline .github/workflows/build-macos-installers.yml:304]: This is a weird thing (that I see also already existed) that.. I dont think does anything in the context of this action?

https://github.com/Chia-Network/actions/blob/main/setjobenv/action.yml

**Review by cmmarslender — CHANGES_REQUESTED** [review]
Left a few requests for minor changes

File: .github/workflows/build-macos-installers.yml (hunk 4 above)
Comment (altendky) [reply .github/workflows/build-macos-installers.yml:303]:
```suggestion
```

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**